server/neptune/gateway/event/push_event_handler.go

```diff
@@ -80,6 +83,16 @@ func (p *PushHandler) handle(ctx context.Context, event Push) error {
 				Name:     event.Repo.Name,
 				Owner:    event.Repo.Owner,
 			},
+			Root: workflows.Root{
+				Name: "TODO",
+				Plan: workflows.Job{
+					Steps: p.generatePlanSteps(event.Repo.ID()),
+				},
+				Apply: workflows.Job{
+					Steps: p.generateApplySteps(event.Repo.ID()),
+				},
+			},
+			TerraformWorkflow: workflows.Terraform,
```

Comment:
requests have to be serializable since they're sent to the temporal server so i don't think this will work.  (function types are not serializable).  I'd probably just pass this in to the function directly in the workflow here if anything:
https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/neptune/workflows/deploy.go#L20

Reply:
ah good point, forgot they need to be serialized